### ISSUE TITLE
Names used in nested functions should not be flagged as unsued

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#9403]
```ts
function f(a) {
    function b() {
        a++;
    }
    b();
}
```

`a` should not be marked us unused

**Comment by aleksey-bykov** [comment]
how is this going to work with 

fiction third<T>(x:any, y: any, z: T) : T { return z; }

**Comment by speigg** [comment]
This does not seem to be fixed yet. I'm seeing the same problem with nested (arrow) functions. 

*This works:*
```ts
const x = 1;
setTimeout(function() { console.log(x) })
```

*This marks x as unused (incorrectly):*
```ts
const x = 1;
setTimeout(() => console.log(x))
```
